main.go

```diff
@@ -1,9 +1,482 @@
+
+func histogram (s []float64) *plot.Plot {
+
+	// builds the histogram
+	// input : s = slice containing the values to plot
+
+	p, err := plot.New()
+	if err != nil {
+		log.Panic(err)
+	}
+
+	val := make(plotter.Values, len(s))
+	for i,_ := range s {
+		val[i] = s[i]
+	}
+
+	p.Title.Text = "Histogram"
+	h, err := plotter.NewHist(val, len(val))
+	if err != nil {
+		log.Panic(err)
+	}
+	p.Add(h)
+
+	return p
+}
+
+/*
+######
+Main function
+######
+*/
+
 func main() {
+
+	// reads the raw datas and format it into a slice of strings, each containing the information for a SN
+	datas, err := ioutil.ReadFile("./data/jla_lcparams.txt")
+	if err != nil {
+		fmt.Println(err)
+	}
+	str := strings.Trim(string(datas), "\n")		// gets rid, if needed, of a new line symbol at the end of the file
+	supernovae := strings.Split(str, "\n")			// builds a slice with data for a SN in each item
+	supernovae = append(supernovae[:0], supernovae[1:]...)	// suppress the first item (labels of the columns)
+
+	// creates slices with the needed values from JLA datas
+
+	sn_names := make([]string, 0)	// names
+	zcmb := make([]float64, 0)	// redshift in the cmb frame
+	zhel := make([]float64, 0)	// redshift in the heliocentric frame
+	mb := make([]float64, 0)	// b band peak magnitude
+	dmb := make([]float64, 0)	// b band peak magnitude error
+	stretch := make([]float64, 0)	// stretch factor
+	dstretch := make([]float64, 0)	// stretch factor error
+	colour := make([]float64, 0)	// colour factor
+	dcolour := make([]float64, 0)	// colour factor error
+	m_stell := make([]float64, 0)	// log10 of the host galaxy stellar mass
+	exp := make([]float64, 0)	// supernova sample identifier : 1 = SNLS, 2 = SDSS, 3 = lowz, 4 = Riess HST
+	ra_jla := make([]float64, 0)	// right ascension (in degrees)
+	de_jla := make([]float64, 0)	// declination (in degrees)
+
+	for _, v := range supernovae {
+		split_str := strings.Split(v, " ")
+		for x, y := range split_str {
+			switch x {
+			case 0:
+				sn_names = append(sn_names, y)
+			case 1:
+				z, _ := strconv.ParseFloat(y, 64)
+				zcmb = append(zcmb, z)
+			case 2:
+				z, _ := strconv.ParseFloat(y, 64)
+				zhel = append(zhel, z)
+			case 4:
+				z, _ := strconv.ParseFloat(y, 64)
+				mb = append(mb, z)
+			case 5:
+				z, _ := strconv.ParseFloat(y, 64)
+				dmb = append(dmb, z)
+			case 6:
+				z, _ := strconv.ParseFloat(y, 64)
+				stretch = append(stretch, z)
+			case 7:
+				z, _ := strconv.ParseFloat(y, 64)
+				dstretch = append(dstretch, z)
+			case 8:
+				z, _ := strconv.ParseFloat(y, 64)
+				colour = append(colour, z)
+			case 9:
+				z, _ := strconv.ParseFloat(y, 64)
+				dcolour = append(dcolour, z)
+			case 10:
+				z, _ := strconv.ParseFloat(y, 64)
+				m_stell = append(m_stell, z)
+			case 17:
+				z, _ := strconv.ParseFloat(y, 64)
+				exp = append(exp, z)
+			case 18:
+				z, _ := strconv.ParseFloat(y, 64)
+				ra_jla = append(ra_jla, z)
+			case 19:
+				z, _ := strconv.ParseFloat(y, 64)
+				de_jla = append(de_jla, z)
+			}
```

Comment:
you may want to add a `default` case to detect for misformated or corrupted data files.
alternatively, check for the size of `split_str` (instead of adding that `default` case.)
then you can drop the `switch` and directly index into `split_str`